CRM/Mailing/BAO/TrackableURL.php

```diff
@@ -66,14 +66,16 @@ private static function getBasicTrackerURL($url, $mailing_id, $queue_id) {
         $hrefExists = TRUE;
       }
 
-      $tracker->url = $url;
-      $tracker->mailing_id = $mailing_id;
+      $params = ['url' => $url, 'mailing_id' => $mailing_id];
+      CRM_Utils_Hook::pre('create', 'TrackableURL', NULL, $params);
```

Comment:
If we replaced this with a call to `writeRecord` it would be more standard & get the hook 'for free'